android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -863,15 +863,39 @@ abstract class EngineSession(
     /**
      * Requests the analysis results for a given product page URL.
      *
-     * @param onSuccess callback invoked if the engine API returns a valid response.
-     * @param onError callback invoked if there was an error getting the response.
+     * @param onResult callback invoked if the engine API returns a valid response.
+     * @param onException callback invoked if there was an error getting the response.
      */
     abstract fun requestProductAnalysis(
         url: String,
         onResult: (ProductAnalysis) -> Unit,
         onException: (Throwable) -> Unit,
     )
 
+    /**
+     * Requests the reanalysis of a product for a given product page URL.
+     *
+     * @param onResult callback invoked if the engine API returns a valid response.
+     * @param onException callback invoked if there was an error getting the response.
+     */
+    abstract fun reanalyzeProduct(
+        url: String,
+        onResult: (String?) -> Unit,
```

Comment:
The callback should return a non nullable String as GV is returning non null when there's a valid response. Same for the function below.